[style-checker] Update warnings for use of `_VERSION_MIN_REQUIRED` and `_VERSION_MAX_ALLOWED` macros to reflect current reality

pxlcoder opened the pull request:
#### b3dac5b6a0a57ece20cd1ea8a6069cfa90a6ba47
<pre>
[style-checker] Update warnings for use of `_VERSION_MIN_REQUIRED` and `_VERSION_MAX_ALLOWED` macros to reflect current reality
<a href="https://bugs.webkit.org/show_bug.cgi?id=286176">https://bugs.webkit.org/show_bug.cgi?id=286176</a>
<a href="https://rdar.apple.com/143160437">rdar://143160437</a>

Reviewed by Alexey Proskuryakov and Elliott Williams.

As of macOS 11, it is no longer the case that `__MAC_OS_X_VERSION_MIN_REQUIRED`
does not work for minor versions. Additionally, non-macOS platforms also support
use of `_VERSION_MIN_REQUIRED` with minor versions.

Tiny versions continue to be unsupported for both macros. Thus, the
`_VERSION_MIN_REQUIRED` is relaxed to check `% 100` for all platforms,
and an equivalent check for `_VERSION_MAX_ALLOWED` is introduced.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_os_version_checks):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest):

Add tests and update existing tests to reflect the minimum support macOS version.

Canonical link: <a href="https://commits.webkit.org/289361@main">https://commits.webkit.org/289361@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9b5902aa343748c46d3f5c46e24e096e5076fa91

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/86675 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6182 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41007 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91520 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37407 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/88724 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6450 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14240 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/67026 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/24809 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/89678 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/4902 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/78475 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47344 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/86173 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4700 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/32806 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36525 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/75148 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/33691 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93415 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/13828 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10025 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/75817 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14029 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74297 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75004 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19312 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/17713 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/6613 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13473 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/13851 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13589 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17033 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15373 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Tools/Scripts/webkitpy/style/checkers/cpp.py:
```diff
@@ -1220,13 +1215,6 @@ def check_os_version_checks(filename, clean_lines, line_number, error):
-    for version_match in _RE_PATTERN_XCODE_MIN_REQUIRED_MACRO.finditer(line):
-        os_prefix = version_match.group(1)
-        version_number = int(version_match.group(2))
-        if os_prefix == '__MAC_OS_X' and version_number % 100 != 0 or os_prefix != '__MAC_OS_X' and version_number % 10000 != 0:
```
I'll preserve the `version_number % 100 != 0` for tiny versions.

aproskuryakov commented on Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
```diff
@@ -6952,9 +6952,9 @@ def test_os_version_checks(self):
         self.assert_lint('#if PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED < 101300', '', 'Source/WTF/wtf/PlatformEnableCocoa.h')
         self.assert_lint('#if PLATFORM(MAC) && __IPHONE_OS_VERSION_MIN_REQUIRED > 120000', '', 'Source/WTF/wtf/PlatformHave.h')
-        self.assert_lint('#if PLATFORM(MAC) && __IPHONE_OS_VERSION_MIN_REQUIRED > 120400', 'Incorrect OS version check. VERSION_MIN_REQUIRED values never include a minor version. You may be looking for a combination of VERSION_MIN_REQUIRED for target OS version check and VERSION_MAX_ALLOWED for SDK check.  [build/version_check] [5]', 'Source/WTF/wtf/PlatformEnableCocoa.h')
+        self.assert_lint('#if PLATFORM(MAC) && __IPHONE_OS_VERSION_MIN_REQUIRED > 120400', '', 'Source/WTF/wtf/PlatformEnableCocoa.h')
```
What is this test even, why does it look at both MAC and IPHONE?

aproskuryakov commented on Tools/Scripts/webkitpy/style/checkers/cpp.py:
```diff
@@ -1197,19 +1197,14 @@ def check_invalid_increment(clean_lines, line_number, error):
-    2. VERSION_MIN_REQUIRED never changes for a minor OS version.
-
-    These should be centralized in the wtf/Platform*.h suite of files.
+    2. These should be centralized in the wtf/Platform*.h suite of files.
```
Not sure how 1 and 2 are different here.

pxlcoder committed:
[style-checker] Update warnings for use of `_VERSION_MIN_REQUIRED` and `_VERSION_MAX_ALLOWED` macros to reflect current reality
https://bugs.webkit.org/show_bug.cgi?id=286176
rdar://143160437

Reviewed by Alexey Proskuryakov and Elliott Williams.

As of macOS 11, it is no longer the case that `__MAC_OS_X_VERSION_MIN_REQUIRED`
does not work for minor versions. Additionally, non-macOS platforms also support
use of `_VERSION_MIN_REQUIRED` with minor versions.

Tiny versions continue to be unsupported for both macros. Thus, the
`_VERSION_MIN_REQUIRED` is relaxed to check `% 100` for all platforms,
and an equivalent check for `_VERSION_MAX_ALLOWED` is introduced.

* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(check_os_version_checks):
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(WebKitStyleTest):

Add tests and update existing tests to reflect the minimum support macOS version.

Canonical link: https://commits.webkit.org/289361@main
